lib/active_merchant/billing/gateways/paybox_direct.rb

```diff
@@ -125,6 +133,11 @@ def add_reference(post, identification)
         post[:numtrans] = identification[10,10]
       end
 
+      def add_amount(post, money, options)
+        post[:montant] = ('0000000000' + (money ? amount(money) : ''))[-10..-1]
```

Comment:
i refactored out the money-type fields into it's own method, from its previous home on `commit`. It looks like this gateway needs a specific length for the `montant` field. Dunno.